lib/plottr_components/src/components/dashboard/account/proOnboarding/Uploading.js

```diff
@@ -0,0 +1,143 @@
+import React, { useEffect, useState } from 'react'
+import PropTypes from 'react-proptypes'
+import { ProgressBar } from 'react-bootstrap'
+import { t } from 'plottr_locales'
+import { StepBody, StepFooter, StepHeader } from '../../../onboarding/Step'
+
+const typeName = {
+  project: t('Project'),
+  template: t('Template'),
+}
+
+const UploadingConnector = (connector) => {
+  const {
+    platform: {
+      file: { basename, doesFileExist, readFileSync },
+      firebase: { saveCustomTemplate },
+      log,
+      useSettingsInfo,
+      appVersion,
+    },
+    pltr: { migrateIfNeeded },
+  } = connector
+
+  const Uploading = ({ nextStep, projects, templates }) => {
+    const [settings] = useSettingsInfo()
+    const [progressMax, setProgressMax] = useState(100)
+    const [progressCurrent, setProgressCurrent] = useState(1)
+    const [currentObj, setCurrentObj] = useState(null)
+
+    useEffect(() => {
+      if (!settings.user || !settings.user.id) return
+      const toUpload = []
+      projects.forEach((pr) => {
+        toUpload.push({ type: 'project', path: pr.path, name: basename(pr.path) })
+      })
+      templates.forEach((tm) => {
+        toUpload.push({ type: 'template', id: tm.id, data: tm, name: tm.name })
+      })
+      // beging uploading
+      setProgressMax(toUpload.length)
+      const allPromises = toUpload.reduce((p, obj, idx) => {
+        return p.then(() => {
+          setCurrentObj(`${typeName[obj.type]}: ${obj.name}`)
+          setProgressCurrent(idx + 1)
+          log.info(typeName[obj.type], 'uploading ...', obj.name)
+          if (obj.type == 'project') {
+            // upload project
+            // uploadProject(obj)
+            return new Promise((resolve, _reject) => {
+              setTimeout(() => resolve(), 100)
+            })
+          } else {
+            return new Promise((resolve, _reject) => {
+              setTimeout(() => resolve(), 100)
+            })
+            // upload template
+            // return saveCustomTemplate(settings.user.id, obj.data)
+          }
+        })
+      }, Promise.resolve())
+      allPromises.then(() => {
+        // done! Go to the next step
+        setCurrentObj(t('Done!'))
+        setTimeout(nextStep, 1500)
+      })
+      // toUpload.forEach((obj, idx) => {
+      //   setCurrentObj(`${typeName[obj.type]}: ${obj.name}`)
+      //   setProgressCurrent(idx + 1)
+      //   log.info(typeName[obj.type], 'uploading ...', obj.name)
+      //   if (obj.type == 'project') {
+      //     // upload project
+      //     uploadProject(obj)
+      //   } else {
+      //     // upload template
+      //     // saveCustomTemplate(settings.user.id, obj.data)
+      //   }
+      // })
+      // // done! Go to the next step
+      // setCurrentObj(t('Done!'))
+      // setTimeout(nextStep, 1500)
+    }, [settings])
+
+    const uploadProject = (projObj) => {
```

Comment:
Now that I'm reading this PR, I have a better idea for uploading the file.
Instead of loading the file and relying on sync, we should use: `overwriteAllKeys(fileId, clientId, state)`
Migration is probably still a good idea.